### PR TITLE
Disregard contributor count prior to article fork

### PR DESCRIPTION
**Pull request by pedrogaudencio**
- [x] only count contributors who made commits after fork creation
- [x] update time format in `--since` flag
- [x] refactor contributor commits check
- [x] fix week-based edge case
- [x] add tests

Closes okTurtles/issues/87

---

<a href="https://app.devin.ai/review/okturtles/forkana/pull/100">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: routers/web/repo/editor.go [hunk 1]
```diff
@@ -233,9 +233,10 @@ func redirectForCommitChoice[T any](ctx *context.Context, parsed *preparedEditor
...
 
-	// Check if the request came from an article-based route
-	// If so, redirect back to the article view instead of the file view
-	if strings.HasPrefix(ctx.Req.URL.Path, "/article/") {
+	// Check if the request came from an edit to the article route
+	// If so, redirect to the article view instead of the file view
+	articleEditPath := util.URLJoin("_edit", util.PathEscapeSegments(parsed.NewBranchName), util.PathEscapeSegments(treePath))
+	if strings.Contains(ctx.Req.URL.Path, articleEditPath) {
 		ctx.JSONRedirect(ctx.Repo.RepoLink)
 		return
```
[the remark this answers is not in the document]
Comment (pedrogaudencio) [reply routers/web/repo/editor.go]: The previous check `strings.HasPrefix(ctx.Req.URL.Path, "/article/")` was not working because `ctx.Req.URL.Path` contains the form action URL during the POST request, not the original page URL. The form action is `/{owner}/{repo}/_edit/{branch}/{path}` (I believe, from `RepoOperationsLink`), which never starts with `/article/`.

The new check matches the `_edit/{branch}/{treePath}` pattern in the POST url. Even tho this does match all file edits (not just article edits), in the Forkana context this makes sense because:
1. all repositories have subjects and `ctx.Repo.RepoLink` returns `/article/{owner}/{subject}`
2. redirecting to the article view after any file edit is intended

If we need to distinguish article edits from regular file edits in the future, we should pass a context flag (e.g. `ctx.Data["IsArticleEdit"]`) from the route handler instead of relying on the url path inspection.

**Comment by taoeffect** [comment]
> It's true, this is out of the scope of "disregard contributor count" but since it's a quick and small fix I added it there. Should I address it to a separate PR @taoeffect? Let me know so that we can merge it prior to this one and this commit is skipped.

No need to do anything, it sounds like you've fixed it here 👍

**Review by taoeffect — APPROVED** [review]
